src/pages/index.js

```diff
@@ -0,0 +1,201 @@
+);
+
+const confirmationPopup = new PopupWithConfirmation(".confirmation-modal");
+confirmationPopup.setEventListeners();
+
+const api = new Api({});
+
+////////////////////////////////////////////////////// functions ///////////////////////////////////////////////////////////
+
+//function that opening the img popup
+function handleImageClick(card) {
+  newImagePopup.open(card);
+}
+
+function createCard(cardData) {
+  return new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButtonClick,
+    handleLikedButtonClick
+  ).getView();
+}
+
+function addCardElement(cardData) {
+  //sectionCards.addItem(createCard(cardData));
+  api
+    .addNewCard(cardData.title, cardData.link, cardData._id)
+    .then((cardData) => {
+      sectionCards.addItem(createCard(cardData));
+      addCardFormValidator.resetValidation();
+      newCardPopup.close();
+    });
+}
+
+function handleProfileFormSubmit(userData) {
+  api
+    .updateUserInfo(userData.name, userData.descripton)
+    .then(() => {
+      userInfo.setUserInfo(userData.name, userData.descripton);
+      newEditPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error in updateUserInfo:", error);
+      if (error instanceof SyntaxError) {
+        console.error("Response body:", error.body);
+      }
+    });
+}
+
+/*
+handleDeleteButtonClick function will be called only after the click of the delete
+icon on the card that we want to delete
+the method will open the popup of the confirm modal and will wait for the submit 
+event , if the submit event is happaning only then the confirmationPopup.setSubmitAction()
+*/
+function handleDeleteButtonClick(card) {
+  confirmationPopup.open(); //opening the popup
+  confirmationPopup.setSubmitAction(() => {
+    confirmationPopup.setDeleteLoading(true); //cahnge the text content of the confirm button
+    api
+      .deleteCard(card._id)
+      .then((result) => {
+        card.removeCardElement(result); //card.js method to remove the card_element
+      })
+      .catch((err) => {
+        console.error(`${err} Failed to delete post.`);
+      })
+      .finally(() => {
+        confirmationPopup.setDeleteLoading(false);
+      });
+  });
+}
+/*
+handleLikedButtonClick funtion will be called after the event click on the like button
+this function is passed to the card class
+at the begining its checking if the isLiked is true or false and by the state
+of the isLiked its decide's wich api method to call 
+after the api call we manualy changing the isLiked atribute of the img with the
+setLike function from the card class that adding or removing the like button
+active class
+*/
+function handleLikedButtonClick(card) {
+  if (card.isLiked) {
+    api.dislikeCard(card._id).then((res) => {
+      card.setLike(res.isLiked);
+    });
+  } else {
+    api.likeCard(card._id).then((res) => {
+      card.setLike(res.isLiked);
+    });
+  }
+}
+
+function upDateAvatar(avatar) {
+  avatarPopup.setLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```